python/src/aac/in_out/parser/_parse_source.py

```diff
@@ -187,6 +187,27 @@ def _read_arch_file_content(arch_file: str) -> str:
     return content
 
 
+def get_definitions_by_root_key(root_key: str, definitions: list[Definition]) -> list[Definition]:
```

Comment:
Woo, it's outside the range of changes so can't post comment at proper location. While going through and manually testing user experience of the various constraints I was working on `check_file` and `check_directory` of the primitives and trying to get them to fail. In doing so I came across other file handling areas of course since accepting, ingesting, and parsing files are primary root functionalities of AaC. 

One such area was the initial acceptance to parse an AaC file which happens above in the `read_arch_file_content()`. There are two errant situations that are expected to be handled here: empty file and improper file type. However, neither of those are actually properly handled. 

I can make a new empty file and get a return of `All AaC constraint checks were successful.` And I can change the file type to something other than `*.yaml` or `*.aac` (tried `*.pdf` and `*.py`) and get a return of `All AaC constraint checks were successful.` 

Went back to main and tested similar file situations and same behavior is experienced of getting a valid notice. Going to file a bug for looking into this.

Reply:
Here is the bug for investigating and resolving why these are not properly handled.
https://github.com/DevOps-MBSE/AaC/issues/766